include/aws/s3/private/s3_auto_ranged_put.h

```diff
@@ -31,22 +31,27 @@ enum aws_s3_auto_ranged_put_request_tag {
 struct aws_s3_auto_ranged_put {
     struct aws_s3_meta_request base;
 
+    /* Useable after the Create Multipart Upload request succeeds. */
+    struct aws_string *upload_id;
+
+    struct {
+        /* Only meant for use in the next_request function, which is never called concurrently. */
+        uint32_t next_part_number;
+    } threaded_next_request_data;
```

Comment:
Trivial: Does it really worth to make another struct around it?

Reply:
I think so--I want to make sure that people are careful of where they change/read it.